url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/AgentConnection.java

```diff
@@ -0,0 +1,1042 @@
+							log.warn(">> connected with: "+agent);
+							// FIXME: Ponieważ klient może jeszcze nie mieć zalogowanego tego agenta to (dostanie to info póżniej)   
+							// EVENT [Agent], Action [Login], Agent [304], Exten [2001], Name [Olsztyńska Antonina]
+							String sAgentLogin = "EVENT [Agent], Action [Login], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], Name ["+agent.getName()+"]";
+							out.println(sAgentLogin);
+							// FIXME:Agenta najprawdopodobniej nie ma także w kolejce 
+							// +INFO Queue 'bluzeczka', Agent '304', Exten '2001', Name: 'Olsztyńska Antonina' (dostanie to info póżniej)
+							String sAgentQueueMember = "+INFO Queue '"+queue.getName()+"', " +
+									"Agent '"+agent.getId()+"', Exten '"+agent.getExtension()+"', Name: '"+agent.getName()+"'";
+							out.println(sAgentQueueMember);
+							
+							//EVENT [Agent], Action [NewCall], Agent [313], Exten [877], Queue [nogawka], CallerID [978]
+							String sNewCall ="EVENT [Agent], Action [NewCall], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], " +
+									"Queue ["+queue.getName()+"], CallerID ["+call.getCallerID()+"], Uniqueid ["+call.getUniqueID()+"],"+
+									"CreationTime ["+call.getCrationTimeFormated()+"]";
+							out.println(sNewCall);
+							//EVENT [Agent], Action [AgentConnect], Agent [304], Exten [2001], Queue [bluzeczka], CallerID [76726470], Holdtime [0], Uniqueid [1246610416.627]
+							String sAgentConnect ="EVENT [Agent], Action [AgentConnect], " +
+									"Agent ["+agent.getId()+"], Exten ["+agent.getExtension()+"], " +
+									"Queue ["+queue.getName()+"], " +
+									"CallerID ["+call.getCallerID()+"], Holdtime [0], Uniqueid ["+call.getUniqueID()+"]," +
+									"CreationTime ["+call.getCrationTimeFormated()+"]";
+							out.println(sAgentConnect);
+						}
+					}
+				}
+			}
+		}
+	}
+	@Override
+	public void run() {
+		runInternal();
+		p.log.info("Thread died");
+	}
+	
+	private void runInternal() {
+		setThreadName();
+		AgentConnection agentDel;
+		Thread mainThread=null;
+		try {
+			Selector selector = Selector.open();
+			out = new ChannelWriter(sChannel, selector);
+			in = new ChannelReader(sChannel);
+			//in = new BufferedReader(new InputStreamReader(sChannel
+			//		.getInputStream()));
+			String inputLine, outputLine;
+
+			String date;
+			
+			GregorianCalendar gc = new GregorianCalendar();
+			dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+			dateFormatsec = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+			date=dateFormat.format(gc.getTime());
+			String check = new String(	Integer.toString(2008) + 
+														"-" + Integer.toString(11) + 
+														"-0" + Integer.toString(4) + 
+														" 0" + Integer.toString(15));
+			if(date.compareTo(check) > 0){
+				this.isSerwis=false; // naleďż˝y zmieniďż˝ na true, jesli jest potrzeba wejďż˝cia w tryb serwisowy
+			}
+			Date licenseExpires=null;
+			if (p.c.licenseExpires!=""){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.